src/Microsoft.Health.Client/OAuth2ClientCredentialProvider.cs

```diff
@@ -48,6 +49,11 @@ protected override async Task<string> BearerTokenFunction(CancellationToken canc
             using HttpResponseMessage tokenResponse = await _httpClient.PostAsync(_oAuth2ClientCredentialConfiguration.TokenUri, formContent, cancellationToken);
 
             var openIdConnectMessage = new OpenIdConnectMessage(await tokenResponse.Content.ReadAsStringAsync());
+            if (openIdConnectMessage.AccessToken == null)
```

Comment:
Should we check this or should we check the httpClient response status code?

Reply:
Updated to check the response code